targets/unity-v2/source/Shared/Public/OneDSEventsAPI.cs

```diff
@@ -183,24 +207,60 @@ internal static async Task<PlayFabResult<TelemetryIngestionConfigResponse>> GetT
                 },
                 error =>
                 {
-                    result = new PlayFabResult<TelemetryIngestionConfigResponse>{Error = error, CustomData = customData};
+                    result = new PlayFabResult<TelemetryIngestionConfigResponse>{Error = new OneDsError
+                    {
+                        HttpCode = error.HttpCode,
+                        HttpStatus = error.HttpStatus,
+                        Error = error.Error,
+                        ErrorMessage = error.ErrorMessage
+                    }, CustomData = customData};
                 });
             });
-
-            await WaitWhile(() => result == null, 100);
+#if TPL_35
+            return Task.Run(() =>
+            {
+                WaitWhile(() => result == null).Await();
+                return result;
+            });
+#else
+            await WaitWhile(() => result == null);
             return result;
+#endif
         }
 
-        public static async Task WaitWhile(Func<bool> condition, int frequency = 25, int timeout = -1)
+        private const int WaitWhileFrequencyDefault = 25;
+        private const int WaitWhileTimeoutDefault = -1;
+#if TPL_35
+        public static Task WaitWhile(Func<bool> condition, int frequency = WaitWhileFrequencyDefault, int timeout = WaitWhileTimeoutDefault)
+        {
+            return Task.Run(() =>
+            {
+                var waitTask = Task.Run(() =>
+                {
+                    while (condition())
+                    {
+                        Task.Delay(frequency).Await();
+                    }
+                });
+
+                if(waitTask != Task.WhenAny(waitTask, Task.Delay(timeout)).Await())
+                    throw new TimeoutException();
+            });
+        }
+#else
+        public static async Task WaitWhile(Func<bool> condition, int frequency = WaitWhileFrequencyDefault, int timeout = WaitWhileTimeoutDefault)
```

Comment:
Should this go in a shared Utils file? It seems fairly generic. It looks like it could be useful in other parts of C# legacy code.